src/apiserver/api_service_impl.cc

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright 2021 4Paradigm
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "apiserver/api_service_impl.h"
+
+#include <memory>
+#include <set>
+#include <string>
+
+#include "apiserver/interface_provider.h"
+#include "apiserver/json_writer.h"
+#include "brpc/server.h"
+
+namespace fedb {
+namespace http {
+
+APIServiceImpl::~APIServiceImpl() = default;
+
+bool APIServiceImpl::Init(const sdk::ClusterOptions& options) {
+    // If cluster sdk is needed, use ptr, don't own it. SQLClusterRouter owns it.
+    auto cluster_sdk = new ::fedb::sdk::ClusterSDK(options);
+    bool ok = cluster_sdk->Init();
+    if (!ok) {
+        LOG(ERROR) << "Fail to connect to db";
```

Comment:
using `WARNING` to avoid `core dump`

Reply:
`ERROR` won't cause core dump, `FATAL` will.